specs/core/0_beacon-chain.md

```diff
@@ -1385,16 +1379,11 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
     # Verify that outstanding deposits are processed up to the maximum number of deposits
     assert len(body.deposits) == min(MAX_DEPOSITS, state.eth1_data.deposit_count - state.eth1_deposit_index)
 
-    for operations, function in (
-        (body.proposer_slashings, process_proposer_slashing),
-        (body.attester_slashings, process_attester_slashing),
-        (body.attestations, process_attestation),
-        (body.deposits, process_deposit),
-        (body.voluntary_exits, process_voluntary_exit),
-        # @process_shard_receipt_proofs
-    ):
-        for operation in operations:
-            function(state, operation)
+    process_operations(body.proposer_slashings, process_proposer_slashing)
+    process_operations(body.attester_slashings, process_attester_slashing)
+    process_operations(body.attestations, process_attestations)
```

Comment:
Are we sure that this works? Can `process_operations` just take a singular _or_ plural function like that?

Reply:
`process_operations` is the name of the calling function here. As currently written, this isn't correct.

I see that there is an embedded function in the phase 1 def, but this function has no access to that

Reply:
Yes, messed up there, will add the embedded function and fix the name to not be a duplicate. Goal here is to not make a client stuck to processing lists of operations, list by list. Instead, a list like the phase 1 attestations should be processable as a whole, in between the other operation types. Open for suggestions, will fix this first.